bolt-sidecar/src/state/execution.rs

```diff
@@ -79,20 +82,26 @@ pub struct ExecutionState<C> {
     /// proposal duties for a single lookahead.
     block_templates: HashMap<Slot, BlockTemplate>,
 
+    max_commitments_per_slot: NonZero<usize>,
+
     /// The state fetcher client.
     client: C,
 }
 
 impl<C: StateFetcher> ExecutionState<C> {
     /// Creates a new state with the given client, initializing the
     /// basefee and head block number.
-    pub async fn new(client: C) -> Result<Self, TransportError> {
+    pub async fn new(
+        client: C,
+        max_commitments_per_slot: NonZero<usize>,
+    ) -> Result<Self, TransportError> {
         Ok(Self {
             basefee: client.get_basefee(None).await?,
             block_number: client.get_head().await?,
             slot: 0,
             account_states: HashMap::new(),
             block_templates: HashMap::new(),
+            max_commitments_per_slot,
```

Comment:
Just a soft note, I skipped doing like this for adding chain_id validation (see https://github.com/chainbound/bolt/pull/126#discussion_r1670270048 ) because looked unclean to me but if we have to do it here then we can consider it there also.

Reply:
but in this case,  we can't fetch `max_commitments_per_slot` from the client, unlike in your PR where we use the client to read the chain_id. I might have misunderstood what you mean though